src/paho/mqtt/client.py

```diff
@@ -3001,8 +2983,7 @@ def handle_mytopic(client, userdata, message):
         if callback is None or sub is None:
             raise ValueError("sub and callback must both be defined.")
 
-        with self._callback_mutex:
-            self._on_message_filtered[sub] = callback
+        self._on_message_filtered[sub] = callback
```

Comment:
This isn't a simple assignment. I'm unsure whether this is safe to remove the lock.